### ISSUE TITLE
DynamoDB delete_item() on a table that does not exist raises ConditionalCheckFailedException instead of ResourceNotFoundException

### ISSUE THREAD

**Issue by k-le** [getmoto/moto#6441]
> ⚠️ This issue has also been [submitted to the /boto/boto3 repository](https://github.com/boto/boto3/issues/3747) and they suggested that it may be an issue with `moto`.

### Describe the bug

Given a DynamoDB table resource that has been deleted, when I call `delete_item()` on that table resource, then the method raises the `ConditionalCheckFailedException` instead of the `ResourceNotFoundException`.

This behavior was found while writing unit tests against a method that uses the `delete_item()` method. The unit tests use:
- `unittest`
- `boto3:1.26.146`
- `moto:4.1.10`

### Expected Behavior

Given a deleted table resource,
when I call `delete_item()` on that resource,
then the **`ResourceNotFoundException`** should be raised.

### Current Behavior

Given a deleted table resource,
when I call `delete_item()` on that resource,
then the **`ConditionalCheckFailedException`** is raised.

### Reproduction Steps

Here's a code snippet:
```python
import unittest

import boto3
from moto import mock_dynamodb

class TestResults(unittest.TestCase):
    def setUp(self):
        """
        Set up DynamoDB mock.
        """

        self.mock_dynamodb = mock_dynamodb()
        self.mock_dynamodb.start()
        self.dynamodb = boto3.resource("dynamodb", region_name="us-east-1")
        self.dynamodb.create_table(
            TableName="Results",
            BillingMode="PAY_PER_REQUEST",
            KeySchema=[
                {
                    "AttributeName": "userId",
                    "KeyType": "HASH"
                },
                {
                    "AttributeName": "resultId",
                    "KeyType": "RANGE"
                }
            ],
            AttributeDefinitions=[
                {
                    "AttributeName": "userId",
                    "AttributeType": "S"
                },
                {
                    "AttributeName": "resultId",
                    "AttributeType": "S"
                }
            ]
        )
        self.results = Results(self.dynamodb)
        self.results_table = self.dynamodb.Table("Results")

    def test_delete_result_exception(self):
        self.results_table.delete()
        self.results_table.wait_until_not_exists()
        self.results_table.delete_item(
            Key={
                "resultId": "1",
                "userId": "abc-def-ghi"
            }
        )
```

#### Stacktrace

```bash
test_results.py::TestResults::test_delete_result_exception FAILED [100%]
tests/shared/models/test_results.py:322 (TestResults.test_delete_result_exception)
self = <test_results.TestResults testMethod=test_delete_result_exception>

    def test_delete_result_exception(self):
        self.results_table.delete()
>       self.results_table.delete_item(
            Key={
                "resultId": "1",
                "userId": "abc-def-ghi"
            }
        )

test_results.py:334: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/boto3/resources/factory.py:580: in do_action
    response = action(self, *args, **kwargs)
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/boto3/resources/action.py:88: in __call__
    response = getattr(parent.meta.client, operation_name)(*args, **params)
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/botocore/client.py:530: in _api_call
    return self._make_api_call(operation_name, kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <botocore.client.DynamoDB object at 0x121cfe650>
operation_name = 'DeleteItem'
api_params = {'Key': {'resultsId': {'S': '1'}, 'userId': {'S': 'abc-def-ghi'}}, 'TableName': 'Results'}

    def _make_api_call(self, operation_name, api_params):
        operation_model = self._service_model.operation_model(operation_name)
        service_name = self._service_model.service_name
        history_recorder.record(
            'API_CALL',
            {
                'service': service_name,
                'operation': operation_name,
                'params': api_params,
            },
        )
        if operation_model.deprecated:
            logger.debug(
                'Warning: %s.%s() is deprecated', service_name, operation_name
            )
        request_context = {
            'client_region': self.meta.region_name,
            'client_config': self.meta.config,
            'has_streaming_input': operation_model.has_streaming_input,
            'auth_type': operation_model.auth_type,
        }
        api_params = self._emit_api_params(
            api_params=api_params,
            operation_model=operation_model,
            context=request_context,
        )
        endpoint_url, additional_headers = self._resolve_endpoint_ruleset(
            operation_model, api_params, request_context
        )
        request_dict = self._convert_to_request_dict(
            api_params=api_params,
            operation_model=operation_model,
            endpoint_url=endpoint_url,
            context=request_context,
            headers=additional_headers,
        )
        resolve_checksum_context(request_dict, operation_model, api_params)
    
        service_id = self._service_model.service_id.hyphenize()
        handler, event_response = self.meta.events.emit_until_response(
            'before-call.{service_id}.{operation_name}'.format(
                service_id=service_id, operation_name=operation_name
            ),
            model=operation_model,
            params=request_dict,
            request_signer=self._request_signer,
            context=request_context,
        )
    
        if event_response is not None:
            http, parsed_response = event_response
        else:
            apply_request_checksum(request_dict)
            http, parsed_response = self._make_request(
                operation_model, request_dict, request_context
            )
    
        self.meta.events.emit(
            'after-call.{service_id}.{operation_name}'.format(
                service_id=service_id, operation_name=operation_name
            ),
            http_response=http,
            parsed=parsed_response,
            model=operation_model,
            context=request_context,
        )
    
        if http.status_code >= 300:
            error_code = parsed_response.get("Error", {}).get("Code")
            error_class = self.exceptions.from_code(error_code)
>           raise error_class(parsed_response, operation_name)
E           botocore.errorfactory.ConditionalCheckFailedException: An error occurred (ConditionalCheckFailedException) when calling the DeleteItem operation: A condition specified in the operation could not be evaluated.

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/botocore/client.py:964: ConditionalCheckFailedException
```

### Environment details (OS name and version, etc.)

macOS 13.4 | Python 3.9

**Comment by rafcio19** [comment]
@bblommers I can take a look

**Comment by bblommers** [comment]
Hey @k-le, thanks for raising this and adding the repro! This should now be fixed as of moto >= 4.1.13.dev7 - just let us know if you run into any other issues.

**Comment by k-le** [comment]
Thanks for the work @rafcio19 and @bblommers, wonderful fix!
